### PR TITLE
toml file for installation, replaces setup.py

### PR DESCRIPTION
**Pull request by martinkilbinger**
## Summary

A new installation file `pyproject.toml` replaces `setup.py`.

Install the code with
`python -m pip install .`.

## Reviewer Checklist

> Reviewers should tick the following boxes before approving and merging the PR.

- [x] The PR is assigned to the developer
- [x] The PR has appropriate labels
- [x] The PR is included in appropriate projects and/or milestones
- [x] The PR includes a clear description of the proposed changes
- [x] If the PR addresses an open issue the description includes "closes #<ISSUE NUMBER>"
- [x] The code and documentation style match the current standards
- [x] Documentation has been added/updated consistently with the code
- [ ] All CI tests are passing
- [x] All changed files have been checked and comments provided to the developer
- [ ] All of the reviewer's comments have been satisfactorily addressed by the developer

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  "pytest-black",
+  "pytest-cov",
+  "pytest-pydocstyle",
+]
+dev = ["pycs[lint,release,test]"]
+
+[project.urls]
+Source = "https://github.com/CosmoStat/cosmostat"
+Documentation = "https://github.com/CosmoStat/cosmostat"
+Tracker = "https://github.com/CosmoStat/cosmostat/issues"
```
Comment (sfarrens) [inline pyproject.toml]: update to cs_util repo

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+line-length = 88
+
+[tool.pydocstyle]
+convention = "numpy"
+# Only run docstyle tests on module init files until documentation cleaned up
+match = "__init__.py"
+
+[tool.pytest.ini_options]
+addopts = "--verbose --black --pydocstyle --cov=pycs"
+testpaths = ["pycs"]
```
Comment (sfarrens) [inline pyproject.toml]: update to cs_util package name

**Review by sfarrens — CHANGES_REQUESTED** [review]

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    "build",
+    "twine",
+]
+test = [
+  "pytest",
+  "pytest-black",
+  "pytest-cov",
+  "pytest-pydocstyle",
+]
+dev = ["pycs[lint,release,test]"]
```
Comment (sfarrens) [inline pyproject.toml]:
```suggestion
dev = ["cs_util[lint,release,test]"]
```

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+[tool.black]
+line-length = 88
+
+[tool.pydocstyle]
+convention = "numpy"
+# Only run docstyle tests on module init files until documentation cleaned up
+match = "__init__.py"
+
+[tool.pytest.ini_options]
+addopts = "--verbose --black --pydocstyle --cov=pycs"
```
Comment (sfarrens) [inline pyproject.toml]:
```suggestion
addopts = "--verbose --black --pydocstyle --cov=cs_util"
```

File: pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+Documentation = "https://github.com/CosmoStat/cosmostat"
+Tracker = "https://github.com/CosmoStat/cosmostat/issues"
+
+[tool.black]
+line-length = 88
+
+[tool.pydocstyle]
+convention = "numpy"
+# Only run docstyle tests on module init files until documentation cleaned up
+match = "__init__.py"
```
Comment (sfarrens) [inline pyproject.toml]: Remove

**Review by sfarrens — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 1 above)
Comment (martinkilbinger) [reply pyproject.toml]: Done

File: pyproject.toml (hunk 5 above)
Comment (martinkilbinger) [reply pyproject.toml]: Done

**Comment by sfarrens** [comment]
Looks good @martinkilbinger, don't forget to run Black to reformat your .py files.

**Comment by sfarrens** [comment]
And remote `setup.py` and `setup.cfg`!

File: cs_util/__init__.py [hunk 6]
```diff
@@ -28,11 +28,11 @@
...
+    _version = version("cs_util")
 except Exception:  # pragma: no cover
-    _version = 'Unkown'
+    _version = "Unkown"
     warn(
-        'Could not extract package metadata. Make sure the package is '
-        + 'correctly installed.',
+        "Could not extract package metadata. Make sure the package is "
+        + "correctly installed.",
     )
```
Comment (sfarrens) [inline cs_util/__init__.py]: Remove

File: cs_util/__init__.py [hunk 7]
```diff
@@ -28,11 +28,11 @@
...
 __version__ = _version
```
Comment (sfarrens) [inline cs_util/__init__.py]:
```suggestion
__version__ = "0.0.0"
```

**Review by sfarrens — CHANGES_REQUESTED** [review]

**Comment by sfarrens** [comment]
Hey @martinkilbinger don't forget to remove the `setup.cfg` and you should also fix the whitespace formatting issue in `calc.py` if you can. Once that is done, I think this PR can be merged.

File: cs_util/__init__.py (hunk 7 above)
Comment (martinkilbinger) [reply cs_util/__init__.py]: I changed it, to 0.3.4 which is the current version.

File: cs_util/__init__.py (hunk 6 above)
Comment (martinkilbinger) [reply cs_util/__init__.py]: Done.

File: cs_util/__init__.py [hunk 8]
```diff
@@ -25,14 +25,4 @@
...
-try:
-    _version = version('cs_util')
-except Exception:  # pragma: no cover
-    _version = 'Unkown'
-    warn(
-        'Could not extract package metadata. Make sure the package is '
-        + 'correctly installed.',
-    )
-__version__ = _version
+__version__ = '0.3.4'
```
Comment (sfarrens) [inline cs_util/__init__.py]:
```suggestion
__version__ = "0.3.4"
```

otherwise Black will complain.

**Review by martinkilbinger — COMMENTED** [review]
OK

**Pull request merged**